### ISSUE TITLE
[MainUI] Unexplained result of filtering items

### ISSUE THREAD

**Issue by lolodomo** [openhab/openhab-webui#538]
## The problem

The results of filtering items include items having apparently no link with the filter.

![image](https://user-images.githubusercontent.com/10186704/99708659-6acdbe80-2a9e-11eb-8553-c0218234d257.png)

## Expected behavior

Having only items matching the filter.

**Comment by lolodomo** [comment]
Question: the filmter is applied to what ? Item name or item label ?

**Comment by kuifje123** [comment]
I also notice this.
It appears that item name and type are concatenated, hence ot finds test (last 2 letters of the name, first 2 of the type).

**Comment by ghys** [comment]
That might just be it 😄

**Comment by ghys** [comment]
Is there something to do or is your question answered?

**Comment by lolodomo** [comment]
Yes  it has to be fixed! 
Why concataining name and type? 
In case a concatenation with the type is required for a reason I don't understand, why not adding a space between name and type?

**Comment by lolodomo** [comment]
@ghys : any chance to fix this one?

**Comment by eikowagenknecht** [comment]
First analysis: The parameter where to search is "search-in" (see https://framework7.io/docs/searchbar.html). It is set correctly:
`search-in=".item-title, .item-subtitle, .item-footer"`

But it is completely ignored (emptying it or changing it to e.g. subtitle only doesn't change anything).

**Comment by eikowagenknecht** [comment]
I used this issue to get into webpack JS debugging and evaluated how the f7 component works.

Result: That code I mentioned was misleading as it is not used for virtual lists (which is the case here). Instead a custom search function is used ant that just concatenated without adding the space between label and type. Interestingly it already added a space between name and label.

Fixed it with PR #726 and also removed the misleading code there.
